src/Components/EntityState.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace Components
+{
+	public interface EntityState
+	{
+		bool TryAdd<Component>(Component component);
+		Component Get<Component>();
```

Comment:
The intent is for the API not to throw an exception for missing components, though I'll probably leave ArgumentNull's in. I will also change the API to Only Add/Get (without the Try) I think.